src/lib/cimb_translator/Config.h

```diff
@@ -8,119 +8,169 @@ namespace cimbar
+		{
+			return active_conf().legacy_mode;
+		}
+
+		static unsigned color_mode()
 		{
-			return 1; // unless we override per-thread?
+			return active_conf().legacy_mode? 0 : 1; // unless we override per-thread?
 		}
 
-		static constexpr unsigned color_bits()
+		static unsigned color_bits()
 		{
-			return GridConf::color_bits;
+			return active_conf().color_bits;
 		}
 
-		static constexpr unsigned symbol_bits()
+		static unsigned symbol_bits()
 		{
-			return GridConf::symbol_bits;
+			return active_conf().symbol_bits;
 		}
 
-		static constexpr unsigned bits_per_cell()
+		static unsigned bits_per_cell()
 		{
-			return color_bits() + symbol_bits();
+			return active_conf().bits_per_cell();
 		}
 
-		static constexpr unsigned ecc_bytes()
+		static unsigned ecc_bytes()
 		{
-			return GridConf::ecc_bytes;
+			return active_conf().ecc_bytes;
 		}
 
-		static constexpr unsigned ecc_block_size()
+		static unsigned ecc_block_size()
 		{
-			return GridConf::ecc_block_size;
+			return active_conf().ecc_block_size;
 		}
 
-		static constexpr unsigned image_size_x()
+		static unsigned image_size_x()
 		{
-			return GridConf::image_size_x;
+			return active_conf().image_size_x;
 		}
 
-		static constexpr unsigned image_size_y()
+		static unsigned image_size_y()
 		{
-			return GridConf::image_size_y;
+			return active_conf().image_size_y;
 		}
 
-		static constexpr unsigned anchor_size()
+		static unsigned anchor_size()
 		{
 			return 30;
 		}
 
 		static constexpr unsigned cell_size()
 		{
-			return GridConf::cell_size;
+			return 8;
 		}
 
-		static constexpr unsigned cell_spacing_x()
+		static unsigned cell_spacing_x()
 		{
-			return GridConf::cell_spacing_x;
+			return active_conf().cell_spacing_x;
 		}
 
-		static constexpr unsigned cell_spacing_y()
+		static unsigned cell_spacing_y()
 		{
-			return GridConf::cell_spacing_y;
+			return active_conf().cell_spacing_y;
 		}
 
-		static unsigned corner_padding_x();
-
-		static unsigned corner_padding_y();
+		static unsigned corner_padding_x()
+		{
+			return active_conf().corner_padding_x();
+		}
 
-		static constexpr unsigned cell_offset()
+		static unsigned corner_padding_y()
 		{
-			return GridConf::cell_offset;
+			return active_conf().corner_padding_y();
 		}
 
-		static constexpr unsigned cells_per_col_x()
+		static unsigned cell_offset()
 		{
-			return GridConf::cells_per_col_x;
+			return active_conf().cell_offset;
 		}
 
-		static constexpr unsigned cells_per_col_y()
+		static unsigned cells_per_col_x()
 		{
-			return GridConf::cells_per_col_y;
+			return active_conf().cells_per_col_x;
 		}
 
-		static unsigned total_cells();
+		static unsigned cells_per_col_y()
+		{
+			return active_conf().cells_per_col_y;
+		}
 
-		static unsigned decode_window_bits();
+		static unsigned total_cells()
+		{
+			return active_conf().total_cells();
+		}
 
-		static unsigned capacity(unsigned bitspercell=0);
+		// leave bitspercell here
+		static unsigned capacity(unsigned bitspercell=0)
```

Comment:
As part of the split symbol/color, 2-stage encode (+decode), we actually do vary bits per cell by how many symbol vs color bits there are. The default will calculate it from `symbol_bits + color_bits`, since we usually want the full count.